catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliFilterDelegate.java

```diff
@@ -0,0 +1,528 @@
+        return filter;
+    }
+
+    @Override
+    public String propertyIsLike(String propertyName, String pattern, boolean isCaseSensitive) {
+        String filter = filterFactory.buildPropertyIsLike(pattern);
+        return filter;
+    }
+
+    // Temporal
+    @Override
+    public String after(String propertyName, Date date) {
+        String filter = EMPTY_STRING;
+        if (isSupportedQueryableAttribute(propertyName)) {
+            filter = propertyIsGreaterThanOrEqualTo(propertyName, date);
+        }
+        return filter;
+    }
+
+    @Override
+    public String before(String propertyName, Date date) {
+        String filter = EMPTY_STRING;
+        if (isSupportedQueryableAttribute(propertyName)) {
+            filter = propertyIsLessThanOrEqualTo(propertyName, date);
+        }
+        return filter;
+    }
+
+    @Override
+    public String during(String propertyName, Date startDate, Date endDate) {
+        String startDateFilter = after(propertyName, startDate);
+        String endDateFilter = before(propertyName, endDate);
+        return and(Arrays.asList(startDateFilter, endDateFilter));
+    }
+
+    //  Spatial
+    @Override
+    public String disjoint(String propertyName, String wkt) {
+        String filter = EMPTY_STRING;
+        if (isSupportedQueryableAttribute(propertyName)) {
+            filter = filterFactory.buildDisjointFilter(propertyName, wkt);
+        }
+        return filter;
+    }
+
+    @Override
+    public String within(String propertyName, String wkt) {
+        String filter = EMPTY_STRING;
+        if (isSupportedQueryableAttribute(propertyName)) {
+            filter = filterFactory.buildWithinFilter(propertyName, wkt);
+        }
+        return filter;
+    }
+
+    @Override
+    public String intersects(String propertyName, String wkt) {
+        String filter = EMPTY_STRING;
+        if (isSupportedQueryableAttribute(propertyName)) {
+            filter = filterFactory.buildIntersectsFilter(propertyName, wkt);
+        }
+        return filter;
+    }
+
+    @Override
+    public String dwithin(String propertyName, String wkt, double distance) {
+        String filter = EMPTY_STRING;
+        if (isSupportedQueryableAttribute(propertyName)) {
+            filter = filterFactory.buildDWithinFilter(propertyName, wkt, distance);
+        }
+        return filter;
+    }
+
+    @Override
+    public String beyond(String propertyName, String wkt, double distance) {
+        String filter = EMPTY_STRING;
+        if (isSupportedQueryableAttribute(propertyName)) {
+            filter = filterFactory.buildBeyondFilter(propertyName, wkt, distance);
+        }
+        return filter;
+    }
+
+    private String getStringFromDate(Date date) {
+        TimeZone timeZone = TimeZone.getTimeZone(UTC);
+        Calendar cal = Calendar.getInstance(timeZone);
+        cal.setTime(date);
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DATE_FORMAT);
```

Comment:
You might be interested in [`FastDateFormat`](http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/time/FastDateFormat.html). It's thread-safe (unlike SimpleDateFormat), so you can create a static member instance and not have to create a new one each time this method is called.